src/steps/check-network-request.ts

```diff
@@ -0,0 +1,115 @@
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+import { URL } from 'url';
+import * as querystring from 'querystring';
+
+const OTHER_REQUEST_METHODS = ['POST', 'PATCH', 'PUT'];
+const VALID_CONTENT_TYPES = ['application/json', 'application/json;charset=UTF-8', 'application/x-www-form-urlencoded'];
+
+export class CheckNetworkRequestStep extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Check for a specific network request';
+  // tslint:disable-next-line:max-line-length
+  protected stepExpression: string = 'there should be (?<reqCount>\\d+) matching network requests? for (?<baseUrl>.+)';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.VALIDATION;
+  protected expectedFields: Field[] = [{
+    field: 'reqCount',
+    type: FieldDefinition.Type.NUMERIC,
+    description: '# of Requests',
+  }, {
+    field: 'baseUrl',
+    type: FieldDefinition.Type.URL,
+    description: 'Base URL Is',
+  }, {
+    field: 'pathContains',
+    type: FieldDefinition.Type.STRING,
+    description: 'Path Contains',
+  }, {
+    field: 'withParameters',
+    type: FieldDefinition.Type.MAP,
+    description: 'Parameters Include',
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const reqCount = stepData.reqCount;
+    const baseUrl = stepData.baseUrl;
+    const pathContains = stepData.pathContains || '';
+    const withParameters = stepData.withParameters;
+
+    try {
+      //// This will ensure that NavigateTo was called
+      await this.client.getCurrentPageInfo('url');
+
+      await this.client.waitForNetworkIdle(10000, false);
+      const requests = await this.client.getFinishedRequests();
+
+      const matchingRequests = requests.filter(r => r.url.startsWith(baseUrl) && r.url.includes(pathContains)); //// string.includes(empty) is always true
```

Comment:
- [ ] Recommend parsing with `new URL()` here too, just in case the `pathContains` provided is something short/trivial that happens to be contained in another part of the URL (e.g. in the query param).

Reply:
should we use `includes` or `equals`? For the next push, I used `equals`.

Reply:
I'd still recommend an `includes`, since the field label indicates `contains`.

It's pretty reasonable to assume a fixed `host`.  It's also reasonable to assume the user may not know the exact path.  For example, the Marketo Munchkin script includes the beta version number of the script as part of the path, which the user has no control over (e.g. `munchkin.marketo.net/156/munchkin.js`).

Which could be reliably checked with `baseUrl=https://munchkin.marketo.net` and `pathContains=/munchkin.js`.

Reply:
Nice